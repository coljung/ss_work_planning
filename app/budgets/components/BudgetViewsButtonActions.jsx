import i18n from 'i18next';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import { Link } from 'react-router';
import { ROUTE_DASHBOARD } from '../../Routes';
import Filter from '../filters/Filter';
import { filterSetup, triggerChange } from '../BudgetViewActions';

class BudgetViewsButtonActions extends Component {
    applyFilters = (filters) => {
        this.props.filterSetup(filters);
    };

    render() {
        return (
            <div className="budgetBtns">
                <Link to={ROUTE_DASHBOARD} >
                    <Button icon="arrow-left">{i18n.t('budgetView.backButton')}</Button>
                </Link>

                <Filter
                    onSave={this.applyFilters}
                    filters={this.props.config}
                />
                <Button disabled={this.props.undoDisabled} onClick={this.props.onUndo} icon="left">{i18n.t('budgetView.undoButton')}</Button>
                <Button disabled={this.props.redoDisabled} onClick={this.props.onRedo} icon="right">{i18n.t('budgetView.redoButton')}</Button>
                <Button onClick={this.props.onExport} icon="export">{i18n.t('budgetView.exportButton')}</Button>
            </div>
        );
    }
}

BudgetViewsButtonActions.propTypes = {
    filterSetup: PropTypes.func.isRequired,
    config: PropTypes.object.isRequired,

    undoDisabled: PropTypes.bool,
    onUndo: PropTypes.func,
    redoDisabled: PropTypes.bool,
    onRedo: PropTypes.func,
    onExport: PropTypes.func,
};

function mapStateToProps(state) {
    const { BudgetViewReducer } = state;
    return {
        config: BudgetViewReducer.config,
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
        filterSetup,
        // triggerChange,
    }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(BudgetViewsButtonActions);
