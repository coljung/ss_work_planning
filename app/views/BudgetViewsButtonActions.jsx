import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Popconfirm } from 'antd';
import { Link } from 'react-router';
import { ROUTE_DASHBOARD } from '../Routes';


export default class BudgetViewsButtonActions extends Component {

    render() {
        return (
            <div>
                <Popconfirm placement="top" title='Create new Budget version'
                    onConfirm={ this.props.saveNew } okText="Yes" cancelText="Cancel">
                    <Button>Save New Version</Button>
                </Popconfirm>
                <Button className="undo" >Save {this.props.currentView} view</Button>

                <Button type="primary">
                    <Link to={ROUTE_DASHBOARD} >
                        Back
                    </Link>
                </Button>
            </div>
        );
    }
}

BudgetViewsButtonActions.propTypes = {
    saveNew: PropTypes.func.isRequired,
    save: PropTypes.func.isRequired,
    currentView: PropTypes.string.isRequired,
};


                // <Button className="undo" >Undo</Button>
                // <Button className="redo" >Redo</Button>
