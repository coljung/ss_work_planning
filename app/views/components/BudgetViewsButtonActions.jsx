import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Popconfirm } from 'antd';
import { Link } from 'react-router';
import { ROUTE_DASHBOARD } from '../../Routes';
import Filter from '../../components/filters/Filter';

export default class BudgetViewsButtonActions extends Component {
    constructor(props) {
        super(props);
        this.state = {
            filterModalActive: false,
        };
    }
    toggleFilterModal = () => {
        this.setState({
            filterModalActive: !this.state.filterModalActive,
        });
    }


    render() {
        const undoDisabled = this.props.undoDisabled ? this.props.undoDisabled : true;
        const redoDisabled = this.props.redoDisabled ? this.props.redoDisabled : true;

        return (
            <div className="budgetBtns">
                <Link to={ROUTE_DASHBOARD} >
                    <Button icon="arrow-left">Back</Button>
                </Link>
                <Popconfirm
                    placement="top"
                    title='Create new Budget version'
                    okText="Yes"
                    cancelText="Cancel"
                    onConfirm={ this.props.saveNew }>
                    <Button type="primary" icon="switcher">Save New Version</Button>
                </Popconfirm>&nbsp;
                <Filter
                    visible={this.state.filterModalActive}
                    onOverlayClick={this.toggleFilterModal.bind(this)} />

                <Button icon="switcher" onClick={this.toggleFilterModal}>Filter</Button>
                <Button disabled={undoDisabled} onClick={this.props.onUndo} icon="left">Undo</Button>
                <Button disabled={redoDisabled} onClick={this.props.onRedo} icon="right">Redo</Button>
            </div>
        );
    }
}

BudgetViewsButtonActions.propTypes = {
    saveNew: PropTypes.func.isRequired,
    undoDisabled: PropTypes.bool,
    onUndo: PropTypes.func,
    redoDisabled: PropTypes.bool,
    onRedo: PropTypes.func,
};
