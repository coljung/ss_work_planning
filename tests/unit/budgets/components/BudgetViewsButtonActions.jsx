import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import BudgetViewsButtonActions from '../../../../app/budgets/components/BudgetViewsButtonActions';

describe('BudgetViewsButtonActions', () => {
    it('should render base layout', () => {
        const output = shallow(
            <BudgetViewsButtonActions saveNew="func">Hello Jest!</BudgetViewsButtonActions>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
