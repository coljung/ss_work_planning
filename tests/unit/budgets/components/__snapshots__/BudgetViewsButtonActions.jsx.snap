// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetViewsButtonActions should render base layout 1`] = `
<div
  className="budgetBtns"
>
  <Link
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/home"
  >
    <Button
      ghost={false}
      icon="arrow-left"
      loading={false}
      prefixCls="ant-btn"
    >
      Back
    </Button>
  </Link>
  <Popconfirm
    cancelText="Cancel"
    okText="Yes"
    okType="primary"
    onConfirm="func"
    placement="top"
    prefixCls="ant-popover"
    title="Create new Budget version"
    transitionName="zoom-big"
    trigger="click"
  >
    <Button
      ghost={false}
      icon="switcher"
      loading={false}
      prefixCls="ant-btn"
      type="primary"
    >
      Save New Version
    </Button>
  </Popconfirm>
  Â 
  <Connect(Filter)
    onOverlayClick={[Function]}
    visible={false}
  />
  <Button
    ghost={false}
    icon="switcher"
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
  >
    Filter
  </Button>
  <Button
    ghost={false}
    icon="left"
    loading={false}
    prefixCls="ant-btn"
  >
    Undo
  </Button>
  <Button
    ghost={false}
    icon="right"
    loading={false}
    prefixCls="ant-btn"
  >
    Redo
  </Button>
</div>
`;
